plugin/src/main/java/org/wildfly/plugin/provision/DeploymentResolution.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.plugin.provision;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Resolve possible deployments from the Maven project dependencies.
+ */
+class DeploymentResolution {
+
+    private static final String WAR = "war";
+    private static final String EAR = "ear";
+    private static final String RAR = "rar";
+    private static final String JAR = "jar";
+    private static final Set<String> SUPPORTED_DEPLOYMENT_TYPES = Set.of(EAR, JAR, RAR, WAR);
+    private static final String TEST_JAR = "test-jar";
+
+    static DeploymentResolution getInstance(MavenProject project) throws MojoExecutionException {
+        return new DeploymentResolution(project);
+    }
+
+    private final Map<String, File> files = new TreeMap<>();
+
+    private DeploymentResolution(MavenProject project) throws MojoExecutionException {
+        for (Artifact artifact : project.getArtifacts()) {
+            if (TEST_JAR.equals(artifact.getType())) {
+                continue;
+            }
```

Comment:
I'm not sure this is needed. The `test-jar` type is not in the `SUPPORTED_DEPLOYMENT_TYPES` so ti doesn't seem like it should be needed.

Reply:
Good catch.